nutrition_labels/grant_tagger.py

```diff
@@ -11,248 +39,358 @@
+
+    Methods
+    -------
+    transform(data, train_data_id)
+        Fit a vectorizer using the text data
+    split_data(X_vect, y)
+        Split the data into train and test sets
+    fit(X, y)
+        Train the model
+    predict(X)
+        Make predictions using the trained model
+    predict_proba(X)
+        Output prediction probabilities using the trained model
+    evaluate(X, y, extra_scores, average)
+        Evaluate various metrics using the model.
+    train_test_info()
+        Create an output dictionary with information about
+        which data points were used in the test or training, and
+        the model predictions for each.
+    save_model(output_path, evaluation_results)
+    load_model(output_path)
+    """
+
     def __init__(
         self,
-        ngram_range=(1,2),
         test_size=0.25,
-        vectorizer_type='count',
-        model_type ='naive_bayes',
-        bert_type = 'bert',
-        relevant_sample_ratio = 1
-        ):
-        self.ngram_range = ngram_range
+        relevant_sample_ratio=1,
+        split_seed=1,
+        vectorizer_type="count",
+        classifier_type="naive_bayes",
+        prediction_cols=["Title", "Grant Programme:Title", "Description"],
+        label_name="Relevance code",
+    ):
         self.test_size = test_size
-        self.vectorizer_type = vectorizer_type
-        self.model_type = model_type
-        self.bert_type = bert_type
         self.relevant_sample_ratio = relevant_sample_ratio
+        self.split_seed = split_seed
+        self.vectorizer_type = vectorizer_type
+        self.classifier_type = classifier_type
+        self.prediction_cols = (*prediction_cols,)
+        self.label_name = label_name
 
-    def transform(self, data):
+    def transform(self, data, train_data_id="Internal ID"):
 
-        if 'Grant texts' not in data:
+        if "Grant texts" not in data:
             # If the training data hasn't come through Prodigy tagging then this won't exist
-            data['Grant texts'] = data[['Title', 'Grant Programme:Title', 'Description']].agg(
-                '. '.join, axis=1
-                )
-        self.X = data['Grant texts'].tolist()
-        y = data['Relevance code']
+            data["Grant texts"] = data[list(self.prediction_cols)].agg(
+                ". ".join, axis=1
+            )
+        self.X_ids = data[train_data_id].tolist()
+        self.X = data["Grant texts"].tolist()
+        self.y = data[self.label_name].tolist()
 
-        if self.vectorizer_type == 'count':
+        if self.vectorizer_type == "count":
             self.vectorizer = CountVectorizer(
-                analyzer='word',
-                token_pattern=r'(?u)\b\w+\b',
-                ngram_range=self.ngram_range
-                )
-        elif self.vectorizer_type == 'tfidf':
+                analyzer="word",
+                token_pattern=r"(?u)\b\w+\b",
+                ngram_range=(1, 2),
```

Comment:
so before that was a param, why is it hard coded now? note btw that you can define either a `params` in the init or a `set_params` method which you can use to pass parameters to the underlying `vectorizer` and `classifier`. all sklearn models implement that interface so you can leverage it and pass in params from the config files that affect the vectorizer and the classifier so you can bette record your experiments.